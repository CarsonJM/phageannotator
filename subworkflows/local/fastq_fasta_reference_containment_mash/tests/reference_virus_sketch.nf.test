nextflow_workflow {

    name "Test Subworkflow: FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH"
    script "subworkflows/local/fastq_fasta_reference_containment_mash/main.nf"
    workflow "FASTQ_FASTA_REFERENCE_CONTAINMENT_MASH"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_fasta_reference_containment_mash"
    tag "reference_virus_sketch_provided"


    test("Should run without failures") {

        when {
            params {
                outdir                  = "$outputDir"
                reference_virus_fasta   = 'https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz'
            }
            workflow {
                """
                input[0] = Channel.of([ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true), file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true) ] ])
                input[1] = Channel.of([ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true) ] ])
                """
            }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.trace.succeeded().size() == 5 },
                { assert snapshot(path("$outputDir/ReferenceIdentification")).match("full_fastq_fasta_reference_containment_mash") }
                )
            }
        }
    }
}
