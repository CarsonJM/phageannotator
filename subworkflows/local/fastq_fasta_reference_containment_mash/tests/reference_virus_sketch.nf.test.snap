{
    "full_fastq_fasta_reference_containment_mash": {
        "content": [
            [
                "all_samples.screen:md5,9cd27acafbf63f1fa92e26c3e83bfa3e",
                [
                    "test.fasta_w_screen_hits.fna.gz:md5,627badd2d06b626af8d7b0b80dc6bfcb"
                ]
            ]
        ],
        "timestamp": "2023-09-12T18:43:19+0000"
    }
}